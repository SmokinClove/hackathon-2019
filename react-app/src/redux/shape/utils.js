export {
  getBoundingBox
}

/**
 * @typedef {{x, y}} Point
 */

 /**
 * @param {Array<Array<Point>>} arrayOfArrayOfPoint
 */
function getBoundingBox(arrayOfArrayOfPoint) {
  const boundingBox = {
    top: Number.POSITIVE_INFINITY,
    left: Number.POSITIVE_INFINITY,
    bottom: Number.NEGATIVE_INFINITY,
    right: Number.NEGATIVE_INFINITY,
  }
  arrayOfArrayOfPoint.forEach(
    arrayOfPoint => arrayOfPoint.forEach(point => {
      if (point.x < boundingBox.left) {
        boundingBox.left = point.x;
      }
      if (point.x > boundingBox.right) {
        boundingBox.right = point.x;
      }
      if (point.y < boundingBox.top) {
        boundingBox.top = point.y;
      }
      if (point.y > boundingBox.bottom) {
        boundingBox.bottom = point.y;
      }
  }));

  return boundingBox;
}
