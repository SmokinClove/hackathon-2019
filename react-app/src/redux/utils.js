export {
  createRequestType
}

/**
 * @param {string} actionType
 */
function createRequestType(actionType) {
  return {
    REQUESTED: `${actionType}_REQUESTED`,
    SUCCESS: `${actionType}_SUCCESS`,
    FAILED: `${actionType}_FAILED`,
  }
}