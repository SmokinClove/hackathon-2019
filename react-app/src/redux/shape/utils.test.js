import { getBoundingBox } from './utils';

describe('getBoundingBox', () => {
  it('shoulds give right output', () => {
    expect(getBoundingBox([])).toMatchSnapshot();
    expect(getBoundingBox([[]])).toMatchSnapshot();
    expect(getBoundingBox([[{x: 1, y: 1}, {x: 2, y: 2}]])).toMatchSnapshot();
    expect(getBoundingBox([[{x: 1, y: 1}, {x: 2, y: 2}], [{x: 0, y: -1}, {x: 100, y: 100}]])).toMatchSnapshot();
    expect(getBoundingBox([[{x: 1, y: 1}, {x: 2, y: 2}], [{x: 0, y: -1}, {x: 100, y: 100}], [{x: -10, y: -1}, {x: 101, y: 102}]])).toMatchSnapshot();
  })
})
