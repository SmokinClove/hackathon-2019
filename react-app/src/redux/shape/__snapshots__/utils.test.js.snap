// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBoundingBox shoulds give right output 1`] = `
Object {
  "bottom": -Infinity,
  "left": Infinity,
  "right": -Infinity,
  "top": Infinity,
}
`;

exports[`getBoundingBox shoulds give right output 2`] = `
Object {
  "bottom": -Infinity,
  "left": Infinity,
  "right": -Infinity,
  "top": Infinity,
}
`;

exports[`getBoundingBox shoulds give right output 3`] = `
Object {
  "bottom": 2,
  "left": 1,
  "right": 2,
  "top": 1,
}
`;

exports[`getBoundingBox shoulds give right output 4`] = `
Object {
  "bottom": 100,
  "left": 0,
  "right": 100,
  "top": -1,
}
`;

exports[`getBoundingBox shoulds give right output 5`] = `
Object {
  "bottom": 102,
  "left": -10,
  "right": 101,
  "top": -1,
}
`;
